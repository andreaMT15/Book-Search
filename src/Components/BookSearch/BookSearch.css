.book-search {
  background-image: url('../../assets/books-1617327_1920.jpg');
  background-size: cover;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100vh;
}

.form-wrapper {
  display: flex;
  justify-content: center;
  width: 100%;
}

.search-input {
  padding: 3px;
  line-height: 2;
  min-width: 600px;
}

#submit-btn {
  background-color: #4b0082;
  color: white;
  font-weight: bold;
  border: 1px solid transparent;
  border-radius: 3px;
  padding: 3px;
  padding-right: 5px;
  line-height: 2;
}
