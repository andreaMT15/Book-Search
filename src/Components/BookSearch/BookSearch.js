import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import './BookSearch.css';

class BookSearch extends Component {
  render() {
    const { handleSubmit } = this.props;
    const { handleChange } = this.props;
    return (
      <div data-testid="book-search" className="search-wrapper">
        <div className="form-wrapper">
          <form className="sb-search-form" onSubmit={handleSubmit}>
            <label htmlFor="text">
              <input
                data-testid="search-input"
                type="text"
                name="searchTerm"
                className="search-input"
                placeholder="Search a book title"
                value={this.props.searchTerm}
                onChange={handleChange}
              />
            </label>

            <Button data-testid="submit-btn" id="submit-btn" type="submit">
              Search
            </Button>
            <div className="error">{this.props.searchError}</div>
          </form>
        </div>
      </div>
    );
  }
}

export default BookSearch;
