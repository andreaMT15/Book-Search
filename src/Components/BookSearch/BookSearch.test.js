import React from 'react';
import { render } from '@testing-library/react';
import BookSearch from './BookSearch';

test('<BookSearch/>', () => {
  const { container } = render(<BookSearch />);
  expect(container).toMatchSnapshot();
});
