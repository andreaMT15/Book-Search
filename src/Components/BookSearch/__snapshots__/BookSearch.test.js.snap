// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookSearch/> 1`] = `
<div>
  <div
    class="book-search"
    data-testid="book-search"
  >
    <div
      class="form-wrapper"
    >
      <form
        class="sb-search-form"
      >
        <label
          for="text"
        >
          <input
            class="search-input"
            data-testid="search-input"
            name="searchTerm"
            placeholder="Search a book title"
            type="text"
            value=""
          />
        </label>
        <button
          class="btn btn-primary"
          data-testid="submit-btn"
          id="submit-btn"
          type="submit"
        >
          Search
        </button>
        <div
          class="error"
          data-testid="input-error"
        />
      </form>
    </div>
  </div>
</div>
`;
